src/test/java/com/openlattice/rehearsal/organization/AssemblerTest.kt

```diff
@@ -1099,6 +1099,80 @@ class AssemblerTest : MultipleAuthenticatedUsersBase() {
 
+    @Test
+    fun testAddRemoveMembers() {
+        /* add read on entity sets 1,2 to user and add as member */
+        // create entity set and materialize in organization
+        val et = createEntityType()
+        val es1 = createEntitySet(et, organizationID)
+        val es2 = createEntitySet(et, organizationID)
+
+        grantMaterializePermissions(organization, es1, setOf())
+        organizationsApi.assembleEntitySets(organizationID, mapOf(es1.id to null))
+        grantMaterializePermissions(organization, es2, setOf())
+        organizationsApi.assembleEntitySets(organizationID, mapOf(es2.id to null))
+
+        // add read on both to user1
+        val readPermission = EnumSet.of(Permission.READ)
+        val es1ReadAcl = Acl(AclKey(es1.id), setOf(Ace(user1, readPermission, OffsetDateTime.MAX)))
+        permissionsApi.updateAcl(AclData(es1ReadAcl, Action.ADD))
+        val es2ReadAcl = Acl(AclKey(es2.id), setOf(Ace(user1, readPermission, OffsetDateTime.MAX)))
+        permissionsApi.updateAcl(AclData(es2ReadAcl, Action.ADD))
+
+        // add user1 as member
+        OrganizationControllerCallHelper.addMemberToOrganization(organizationID, user1.id)
+        Thread.sleep(5000L) // wait for configuring the user
+
+        // both entity sets should be selectable by user1
+        loginAs("user1")
+        val materializedViewAccount = principalApi.materializedViewAccount
+        loginAs("admin")
+
+        val connectionProperties = Properties()
+        connectionProperties["jdbcUrl"] = "jdbc:postgresql://localhost:5432"
+        connectionProperties["username"] = materializedViewAccount.username
+        connectionProperties["password"] = materializedViewAccount.credential
+        connectionProperties["maximumPoolSize"] = 5
+        connectionProperties["connectionTimeout"] = 60000
+
+        // connect with user1(simple member) credentials
+        val user1OrganizationDataSource = TestAssemblerConnectionManager.connect(
+                organizationID,
+                Optional.of(connectionProperties)
+        )
+
+        user1OrganizationDataSource.connection.use { connection ->
+            connection.createStatement().use { stmt ->
+                stmt.executeQuery(TestAssemblerConnectionManager.selectFromEntitySetSql(es1.name))
+                stmt.executeQuery(TestAssemblerConnectionManager.selectFromEntitySetSql(es2.name))
+                stmt.executeQuery(TestAssemblerConnectionManager.selectEdgesOfEntitySetsSql())
+            }
+        }
+
+
+        /* remove membership and remove read on entity set 1 and re-add as member */
+        OrganizationControllerCallHelper.removeMemberFromOrganization(organizationID, user1.id)
+        Thread.sleep(5000L) // wait for configuring the user
+
+        permissionsApi.updateAcl(AclData(es1ReadAcl, Action.REMOVE))
+
+        OrganizationControllerCallHelper.addMemberToOrganization(organizationID, user1.id)
+        Thread.sleep(5000L) // wait for configuring the user
```

Comment:
Do these really end up taking 5 seconds? Seems like a lot of extra time to wait for tests 😢

Reply:
I can try to refine it, but this seemed to work consistently.

Reply:
Is this waiting for a background job to run? Is there a way to call whatever is needed that will block so that we can avoid this?

Reply:
Nope, this is only to make sure, that everything finishes, when adding the member (`MembersAddedToOrganizationEvent` get's fired -> get all entity sets + get all properties of entity sets where user has read  -> grant select for all of them + configure user in database).
Maybe my stack is slow locally, this amount of time was always enough, but probably can be less.

Reply:
I cannot test it right now, because stack is broken, but will try after all the merges

Reply:
Brought it down to 1 sec